examples/minimal/packages/client-react-external-wallet/src/mud/systemCalls.ts

```diff
@@ -1,15 +1,63 @@
-import { type WalletClient } from "viem";
-import { type MUDNetwork } from "./NetworkContext";
+import { parseEther, type Hex } from "viem";
+import { resourceToHex } from "@latticexyz/common";
+import { encodeSystemCallFrom } from "@latticexyz/world";
+import { type ExternalWalletClient } from "./NetworkContext";
+import { type PublicClient, type BurnerClient } from "./setupNetwork";
 import IWorldAbi from "contracts/out/IWorld.sol/IWorld.abi.json";
 
-export const increment = async (client: WalletClient, network: MUDNetwork) => {
-  const { request } = await network.publicClient.simulateContract({
-    address: network.worldAddress,
+// Executes the `increment` system call on behalf of the external wallet using the burner wallet.
+// This is solely for demonstrating burner wallet delegation, as the `increment` call can actually be made by any sender.
+export const increment = async (
+  worldAddress: Hex,
+  publicClient: PublicClient,
+  externalWalletClient: ExternalWalletClient,
+  burnerClient: BurnerClient
+) => {
+  const { request } = await burnerClient.simulateContract({
+    account: burnerClient.account,
+    address: worldAddress,
     abi: IWorldAbi,
-    functionName: "increment",
-    args: [],
-    account: client.account,
+    functionName: "callFrom",
+    args: encodeSystemCallFrom({
+      from: externalWalletClient.account.address,
+      abi: IWorldAbi,
+      functionName: "increment",
+      args: [],
+      systemId: resourceToHex({ type: "system", namespace: "", name: "increment" }),
+    }),
   });
-  const tx = await client.writeContract(request);
-  network.waitForTransaction(tx);
+
+  const tx = await burnerClient.writeContract(request);
+
+  await publicClient.waitForStoreSync(tx);
+};
+
+// TODO: Should use `std-delegations` to limit authority
+const UNLIMITED_DELEGATION = resourceToHex({ type: "system", namespace: "", name: "unlimited" });
+
+export const isDelegated = (delegationControlId: Hex) => delegationControlId === UNLIMITED_DELEGATION;
+
+export const delegateToBurner = async (
+  worldAddress: Hex,
+  publicClient: PublicClient,
+  externalWalletClient: ExternalWalletClient,
+  burnerClient: BurnerClient
+) => {
+  const { request } = await publicClient.simulateContract({
+    account: externalWalletClient.account,
+    address: worldAddress,
+    abi: IWorldAbi,
+    functionName: "registerDelegation",
+    args: [burnerClient.account.address, UNLIMITED_DELEGATION, "0x0"],
+  });
+
+  const tx1 = await externalWalletClient.writeContract(request);
```

Comment:
I think we shouldn't include a transaction simulation within the standard viem client action so that users can have control. If we want to do something advanced like #2081, we can build the functionality outside the standard viem action.